hoomd/filter/filter_.py

```diff
@@ -1,4 +1,7 @@
-class _ParticleFilter:
+from hoomd._hoomd import ParticleFilter
+
+
+class _ParticleFilter(ParticleFilter):
```

Comment:
Import `collections.abc.Hashable` and change this to 
```suggestion
class _ParticleFilter(Hashable, ParticleFilter):
```
then you can get rid of the lines below.

Reply:
I'm getting a `TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases` if I try to inherit Hashable.

Reply:
@danevans09 revert the change, this won't work with pybind.

Reply:
@b-butler Is this actually a pybind problem? I didn't think through that aspect, and it's probably not worth the hassle, but naively I would still expect something like this to work:
```
class _ParticleFilterMetaclass(type(Hashable), type(ParticleFilter)):
    pass

class _ParticleFilter(Hashable, ParticleFilter, metaclass=_ParticleFilterMetaclass):
...
```

Does that kind of metaclass hackery not play nicely with pybind? This is more for my edification, it's probably more complexity than is merited for this problem since `Hashable` is such a minimal `abc` anyway.

Reply:
@b-butler It may or it may not. pybind does not seemingly document this very well. They have a custom metaclass called `pybind11_builtins.pybind11_type` which is used. They don't specify if they overwrite `__new__` or `__init__` although I suspect they do for at least `__init__`. This lack of documentation makes it impossible to know how to properly create the hybrid metaclass (i.e. do we need to create our own `__new__` and `__init__` that we use to call `super().__new__` and such.

They do not document how to use metaclasses from Python with their metaclass, and I do not know if it is stable to rely on this even if it works. We could probably make this work through trial and error, but then is that something to trust will be stable.

Reply:
Lol you tagged yourself instead of me. But OK I get the picture, basically it might work or it might not, and not worth relying on it if pybind isn't making any promises. Fine by me. It was inevitable that there would be some gotchas in trying to inherit C++ classes in Python.

Reply:
Yes, yes I did...